smart_contracts/contract/src/contract_api/runtime.rs

```diff
@@ -404,6 +405,44 @@ pub fn get_call_stack() -> Vec<CallStackElement> {
     bytesrepr::deserialize(bytes).unwrap_or_revert()
 }
 
+/// Manages a message topic.
+pub fn manage_message_topic(
+    topic_name: &str,
+    operation: MessageTopicOperation,
+) -> Result<(), ApiError> {
+    if topic_name.is_empty() {
+        return Err(ApiError::InvalidArgument);
+    }
+
+    let (topic_name_ptr, topic_name_size, _bytes) = contract_api::to_ptr(topic_name);
+    let (operation_ptr, operation_size, _bytes) = contract_api::to_ptr(operation);
```

Comment:
To be fair, looking at the definition this could be passed as a number to the host - unless you plan to add more variants with fields etc. in the future. Adding `#[derive(ToPrimitive, FromPrimitive)]` to `MessageTopicOperation` could help with convenience

Reply:
Indeed, now we could pass it as a plain number, and it would probably work for a new operation like `Remove`. Though my thinking was that if we're going to support more complex operations in the future like `Rename`, or to allow selection of the checksum algorithm we would need space for some parameters.